### PR TITLE
Check serialized temporary files memory is non-empty

### LINKED ISSUE #1001: Crash because of empty /var/cache/libdnf5/temporary_files.toml file?
I just tried my routine `dnf5 update --refresh`:

```
...
After this operation 66 KiB will be used (install 781 MiB, remove 781 MiB).
Is this ok [y/N]: y
Failed to download packages
 An error occurred when parsing the temporary files memory file at '/var/cache/libdnf5/temporary_files.toml': [error] toml::value::as_array(): bad_cast to array
 --> unknown file
   |
 1 | unknown line
   |  the actual type is empty
```

It was reproducible, and work-around was to remove the `temporary_files.toml` file.  DNF tooling:

```
dnf5                 x86_64       5.1.6-20231106004845.10.g8598828e.fc40  copr:copr.fedorainfracloud.org:rpmsoftwaremanagement:dnf5-unstable                 1.3 MiB
dnf5-plugins         x86_64       5.1.6-20231106004845.10.g8598828e.fc40  copr:copr.fedorainfracloud.org:rpmsoftwaremanagement:dnf5-unstable               578.9 KiB
libdnf5              x86_64       5.1.6-20231106004845.10.g8598828e.fc40  copr:copr.fedorainfracloud.org:rpmsoftwaremanagement:dnf5-unstable                 2.5 MiB
libdnf5-cli          x86_64       5.1.6-20231106004845.10.g8598828e.fc40  copr:copr.fedorainfracloud.org:rpmsoftwaremanagement:dnf5-unstable               530.1 KiB
```

### PR DESCRIPTION
**Pull request by jan-kolarik**
Based on the https://github.com/rpm-software-management/dnf5/issues/1001, adding a check after the toml serialization of temporary files memory to make sure the result is not an empty file. Such a file would not be parsable when loading from the disk afterwards.

Closes #1001.

### CODE REVIEW

File: test/libdnf5/repo/test_temp_files_memory.cpp [hunk 1]
```diff
@@ -79,6 +79,15 @@ void TempFilesMemoryTest::test_add_files_when_empty_storage() {
     CPPUNIT_ASSERT_EQUAL(new_paths, memory.get_files());
 }
 
+void TempFilesMemoryTest::test_add_no_files_when_empty_storage() {
```
Comment (j-mracek) [inline test/libdnf5/repo/test_temp_files_memory.cpp:82]: May I ask you to add additional information to test - what is in commit message is perfect - test that it does not crash? Without that information the test looks like that it is not required.

**Comment by j-mracek** [comment]
LGTM, only asking about a comment for the test

File: test/libdnf5/repo/test_temp_files_memory.cpp (hunk 1 above)
Comment (jan-kolarik) [reply test/libdnf5/repo/test_temp_files_memory.cpp:82]: I've added a comment there. Is it OK like that?

Comment (jan-kolarik) [reply test/libdnf5/repo/test_temp_files_memory.cpp:82]: And thanks for the suggestion!

Comment (j-mracek) [reply test/libdnf5/repo/test_temp_files_memory.cpp:82]: Thank you

**Pull request merged**